src/main/resources/db/migration/0200/V240__BackfillSeedbankWithdrawalsBatchId.sql

```diff
@@ -0,0 +1,17 @@
+UPDATE seedbank.withdrawals withdrawals
+SET batch_id = (
+    SELECT batchHistory.batch_id
+    FROM nursery.batch_quantity_history batchHistory
+    INNER JOIN nursery.batches batches
+        ON batches.id = batchHistory.batch_id
+    INNER JOIN seedbank.accessions accessions
+        ON accessions.id = withdrawals.accession_id
+    WHERE batches.species_id = accessions.species_id
+        AND batchHistory.germinating_quantity = withdrawals.withdrawn_quantity
+        AND batchHistory.created_by = withdrawals.created_by
+        AND batchHistory.created_time
+            BETWEEN withdrawals.created_time
+            AND (withdrawals.created_time::date + '1 second'::interval)
```

Comment:
Won't casting to a date cause this to evaluate to 1 second past midnight, which will almost always be earlier than the creation time? Does `withdrawals.created_time + '1 second'::interval` not work?

How have you been testing this? I'm surprised this `BETWEEN` clause works for anything but midnight creation times.

Reply:
I've been testing with my local database, clearing batch IDs and running the query. I thought it was working as expected because postgres was responding with `UPDATED 41` (41 being the number of rows I expected to update). After looking a bit closer, you are right, it was not working.